services/catalog/src/simcore_service_catalog/services/director.py

```diff
@@ -27,29 +27,38 @@
 )
 
 
+class UnresponsiveService(RuntimeError):
+    pass
+
+
 async def setup_director(app: FastAPI) -> None:
     if settings := app.state.settings.CATALOG_DIRECTOR:
         # init client-api
-        logger.debug("Setup director at %s...", settings.base_url)
-        director_client = DirectorApi(base_url=settings.base_url, app=app)
+        logger.debug("Setup director at %s ...", f"{settings.base_url=}")
+        client = DirectorApi(base_url=settings.base_url, app=app)
 
         # check that the director is accessible
-        async for attempt in AsyncRetrying(**director_startup_retry_policy):
-            with attempt:
-                if not await director_client.is_responsive():
-                    raise ValueError("Director-v0 is not responsive")
+        try:
+            async for attempt in AsyncRetrying(**director_startup_retry_policy):
+                with attempt:
+                    if not await client.is_responsive():
+                        raise UnresponsiveService("Director-v0 is not responsive")
 
-                logger.info(
-                    "Connection to director-v0 succeded [%s]",
-                    json_dumps(attempt.retry_state.retry_object.statistics),
-                )
+                    logger.info(
+                        "Connection to director-v0 succeded [%s]",
+                        json_dumps(attempt.retry_state.retry_object.statistics),
+                    )
+        except UnresponsiveService:
+            await client.close()
+            raise
 
-        app.state.director_api = director_client
+        app.state.director_api = client
 
 
 async def close_director(app: FastAPI) -> None:
-    if director_client := app.state.director_api:
-        await director_client.delete()
+    client: Optional[DirectorApi]
```

Comment:
what is the point of declaring this variable?

Reply:
this way we can get autocompletion, search,  etc in vscode